Code/Framework/AzToolsFramework/AzToolsFramework/API/ToolsApplicationAPI.h

```diff
@@ -605,6 +605,18 @@ namespace AzToolsFramework
          * system simulates its execution, and avoids some code duplication.
          */
         virtual void RunRedoSeparately(UndoSystem::URSequencePoint* redoCommand) = 0;
+
+        /**
+         * Create a prefab out of the entities provided, at the path provided.
+         * Automatically detects descendants of entities, and discerns between entities and child instances.
+         */
+        virtual bool CreatePrefab(
+            const AZStd::vector<AZ::EntityId>& entityIds, const char* filePath, bool saveToDisk) = 0;
+
+        /**
+         * Instantiate a prefab from a prefab file.
+         */
+        virtual bool InstantiatePrefab(const char* filePath, AZ::EntityId parent, const AZ::Vector3& position) = 0;
```

Comment:
Instead of returning a bool, I think it is way more useful to return an EntityId of the container, or a list of entities of the prefab. This makes it easier to then do things with the prefab.
You could even create and expose kind of struct that contains these aswell

Reply:
Modified the function to return container entity id instead.